src/components/Breakpoints.js

```diff
@@ -72,54 +94,173 @@ const Breakpoints = React.createClass({
     this.props.removeBreakpoint(breakpoint.location);
   },
 
+  pauseExceptionModeToggled(event) {
+    const { pauseOnExceptions, exceptionPauseModes } = this.props;
+    const targetMode = exceptionPauseModes.filter(
+      item => item.mode === event.target.value
+    )[0];
+
+    pauseOnExceptions(targetMode.shouldPause, targetMode.shouldIgnoreCaught);
+  },
+
+  renderPauseExecutionButton() {
+    const { breakOnNext, isWaitingOnBreak } = this.props;
+
+    return (
+      <button
+        className="pause-execution"
+        disabled={ isWaitingOnBreak }
+        onClick={ breakOnNext }
+      >
+        <Svg name="pause" />
+        { isWaitingOnBreak ?
+          "Will pause on next execution" :
+          "Pause on next execution" }
+      </button>
+    );
+  },
+
+  renderGlobalBreakpoints() {
+    const currentMode = this.props.currentExceptionPauseMode;
+    const _createToggle = (fromMode) => {
+      return (
+        <label className="breakpoint" key={ fromMode.mode }>
+          <input
+            type="radio"
+            onChange={ this.pauseExceptionModeToggled }
+            value={ fromMode.mode }
+            checked={ currentMode.mode === fromMode.mode }
+          />
+          <div className="breakpoint-label">{ fromMode.label }</div>
+        </label>
+      );
+    };
+
+    return (
+      <details>
+        <summary className="_header">
+          { `Exceptions - Pausing on: ${currentMode.headerLabel}` }
+          </summary>
+        { this.props.exceptionPauseModes.map(_createToggle) }
+      </details>
+    );
+  },
+
+  renderUserBreakpointsHeader() {
+    const { toggleAllBreakpoints, breakpointsDisabled,
+      breakpoints, breakpointsLoading } = this.props;
+    const label = breakpointsDisabled ? L10N.getStr("breakpoints.enable") :
+      L10N.getStr("breakpoints.disable");
+    const isIndeterminate = !breakpointsDisabled &&
+      breakpoints.some(x => x.disabled);
+    const clearAll = () => {
+      if (confirm("Are you sure you want to remove all breakpoints?")) {
+        breakpoints.forEach(
+          breakpoint => this.props.removeBreakpoint(breakpoint.location)
+        );
+      }
+    };
+    const toggleAll = () => {
+      toggleAllBreakpoints(!breakpointsDisabled);
+    };
+    const toggleIndeterminate = (input) => {
+      if (input) {
+        input.indeterminate = isIndeterminate;
+      }
+    };
```

Comment:
I like pulling these functions out of the curlies. I could see the markup becoming pretty hard to follow with these.

Personally, I prefer the old JS style for these in line functions, but ... what can you do :)